packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -57,189 +61,79 @@ export class ConflictDetector {
-    try {
-      const zip = new AdmZip(unpackagedZipFile);
-      zip.extractAllTo(tempMetadataPath, true);
-    } catch (error) {
-      this.reportError('error_extracting_org_source', error);
-      return Promise.reject();
-    }
-
-    // 4: convert org source to decomposed (source) format
-    const unconvertedSourcePath = path.join(tempMetadataPath, 'unpackaged');
-    const convertedSourcePath = path.join(tempMetadataPath, 'converted');
-    await this.executeCommand(
-      this.buildMetadataApiConvertOrgSourceCommand(
-        unconvertedSourcePath,
-        convertedSourcePath
-      ),
-      projectPath,
-      cancellationTokenSource,
-      cancellationToken
-    );
-
-    // 5: diff project directory (local) and retrieved directory (remote)
-    // Assume there are consistent subdirs from each root i.e. 'main/default'
-    const localSourcePath: string = path.join(projectPath, data.packageDir);
-    const diffs = this.differ.diff(localSourcePath, convertedSourcePath);
-
-    return diffs;
+    return this.diffs;
   }
 
-  public async executeCommand(
-    command: Command,
-    projectPath: string,
-    cancellationTokenSource: vscode.CancellationTokenSource,
-    cancellationToken: vscode.CancellationToken
-  ): Promise<string> {
-    const startTime = process.hrtime();
-    const execution = new CliCommandExecutor(
-      command,
-      { cwd: projectPath, env: { SFDX_JSON_TO_STDOUT: 'true' } },
+  public createCacheExecutor(
+    username: string,
+    callback: MetadataCacheCallback
+  ): MetadataCacheExecutor {
+    const executor = new MetadataCacheExecutor(
+      username,
+      nls.localize('conflict_detect_execution_name'),
+      'conflict_detect',
+      callback,
       true
-    ).execute(cancellationToken);
-
-    const result = new CommandOutput().getCmdResult(execution);
-    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
-    execution.processExitSubject.subscribe(() => {
-      telemetryService.sendCommandEvent(execution.command.logName, startTime);
-    });
-    return result;
-  }
-
-  protected attachExecution(
-    execution: CommandExecution,
-    cancellationTokenSource: vscode.CancellationTokenSource,
-    cancellationToken: vscode.CancellationToken
-  ) {
-    channelService.streamCommandOutput(execution);
-    channelService.showChannelOutput();
-    notificationService.reportCommandExecutionStatus(
-      execution,
-      cancellationToken
     );
-    ProgressNotification.show(execution, cancellationTokenSource);
-    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+    return executor;
   }
 
-  private clearDirectory(dirToRemove: string, throwErrorOnFailure: boolean) {
-    try {
-      shell.rm('-rf', dirToRemove);
-    } catch (error) {
-      this.reportError('error_cleanup_temp_files', error);
-      if (throwErrorOnFailure) {
-        throw error;
-      }
+  private async handleCacheResults(
```

Comment:
Where is this handleCacheResults used and how is this different from the one in forceSourceDiff.ts ?

Reply:
Never mind, so this is specifically used for the conflict detection part here. I think we need a separate diffService.ts and move some of the diff specific code there. Can be part of a separate devchoice/refactoring story.

Reply:
Agreed. There is some overlap, but they will diverge as conflict detection progresses.